### PR TITLE
feat(jtk): add VISIBILITY column to comments list --extended

### LINKED ISSUE #288: jtk comments: add VISIBILITY column to list --extended
Parent: #230

### `jtk comments list <key> --extended`
**Current:** missing `VISIBILITY` column.
**Expected:**
```
ID | AUTHOR | CREATED | UPDATED | VISIBILITY | BODY
21242 | Aaron Wong | 2026-04-16T09:56:22+0000 | 2026-04-16T09:56:22+0000 | - | Short audit conclusion...
```

`VISIBILITY` is `-` when unrestricted; shows the restriction group/role otherwise.

See #230 for full spec.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Adds `VISIBILITY` column to `jtk comments list --extended` (table and detail modes)
- Maps to Jira API `visibility` object (`type` + `value` fields) on comments
- Shows `-` when unrestricted, restriction group/role name otherwise
- Handles edge case of empty visibility value via `OrDash`

Closes #288

## Test plan
- [x] Presenter spec-lock tests verify exact header order for both default and extended modes (table + detail)
- [x] Dedicated visibility test covers nil, non-nil, and empty-value cases
- [x] Command-level extended test verifies VISIBILITY header and values in output
- [x] `--fields VISIBILITY` projection test verifies column selection works
- [x] `make build && make test && make lint` all pass

### CODE REVIEW

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- None.

**Minor**
- `tools/jtk/internal/present/comment.go`: PR summary says detail mode is supported, but tests only prove table-mode visibility values. Add a `PresentListFull(..., true)` or `comments list --fulltext --extended` assertion that `Visibility` renders `-` / `Administrators`.

**Nit**
- `tools/jtk/internal/cmd/comments/comments_fields_test.go`: the new test comment says `--fields VISIBILITY` works “without requiring --extended”, but the test sets `opts.Extended = true`. Reword the comment or add the no-extended case if that behavior is intended.

**Comment by rianjs** [comment]
## TDD Assessment — Comments Visibility Column

**No blocking issues.** Three-layer test structure is present.

### Gaps found and disposition:
1. **No `--fulltext --extended` cmd-level test** — Added `TestRunList_FullTextExtended_VisibilityField`. ✅ Fixed
2. **Nil visibility not asserted as `-`** — Strengthened `TestRunList_Extended_VisibilityColumn` to check first data row contains `-`. ✅ Fixed
3. **Default-mode spec-lock omits Visibility** — By design; Visibility is Extended-only. The spec-lock correctly excludes it in default mode. No action needed.
4. **No block-mode `--fields Visibility` projection** — Low-priority; table-mode `--fields` test + detail-mode presenter test provide sufficient coverage. Accepted as-is.

File: tools/jtk/internal/present/comment.go [hunk 1]
```diff
@@ -176,6 +181,13 @@ func (CommentPresenter) PresentDeleted(commentID, issueKey string) *present.Outp
 	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/comment.go:181]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): formatVisibility renders only v.Value and discards v.Type ('role' vs 'group'). A role named 'Administrators' and a group named 'Administrators' produce identical output, making the restriction type invisible to users. Consider showing 'role:Administrators' or 'group:Administrators' to preserve the distinction Jira exposes.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/comments/comments_test.go [hunk 2]
```diff
@@ -337,6 +337,68 @@ func TestRunList_NoComments(t *testing.T) {
...
+		},
+	}
+	server := newTestCommentsServer(t, comments)
+	defer server.Close()
+
+	opts, stdout, _ := newCommentsOpts(t, server)
+	opts.Extended = true
+	err := runList(context.Background(), opts, "TEST-1", 50, false, "")
+	testutil.RequireNoError(t, err)
+
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/comments/comments_test.go:362]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): The nil-visibility assertion checks strings.Contains(lines[1], "-") but lines[1] is likely a table-separator row or contains date-formatting dashes (e.g. '2024-01-15'), so the assertion passes trivially regardless of whether VISIBILITY renders as '-'. The test should assert on the specific cell value in the data row.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/comments/comments_test.go [hunk 3]
```diff
@@ -337,6 +337,68 @@ func TestRunList_NoComments(t *testing.T) {
...
+	opts, stdout, _ := newCommentsOpts(t, server)
+	opts.Extended = true
+	err := runList(context.Background(), opts, "TEST-1", 50, true, "")
+	testutil.RequireNoError(t, err)
+
+	output := stdout.String()
+	testutil.Contains(t, output, "Visibility:")
+	testutil.Contains(t, output, "Administrators")
+	testutil.Contains(t, output, "-")
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/comments/comments_test.go]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): testutil.Contains(t, output, "-") in TestRunList_FullTextExtended_VisibilityField trivially passes because every timestamp in the detail output contains hyphens. This does not verify the nil-visibility comment renders its Visibility field as '-'; should assert on the label+value pair, e.g. 'Visibility: -'.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `8583266`

> **Approved** with 3 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 3 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (3 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/comment.go:181`
> formatVisibility renders only v.Value and discards v.Type ('role' vs 'group'). A role named 'Administrators' and a group named 'Administrators' produce identical output, making the restriction type invisible to users. Consider showing 'role:Administrators' or 'group:Administrators' to preserve the distinction Jira exposes.

**💡 Suggestion** - `tools/jtk/internal/cmd/comments/comments_test.go:362`
> The nil-visibility assertion checks strings.Contains(lines[1], "-") but lines[1] is likely a table-separator row or contains date-formatting dashes (e.g. '2024-01-15'), so the assertion passes trivially regardless of whether VISIBILITY renders as '-'. The test should assert on the specific cell value in the data row.

**💡 Suggestion** - `tools/jtk/internal/cmd/comments/comments_test.go:399`
> testutil.Contains(t, output, "-") in TestRunList_FullTextExtended_VisibilityField trivially passes because every timestamp in the detail output contains hyphens. This does not verify the nil-visibility comment renders its Visibility field as '-'; should assert on the label+value pair, e.g. 'Visibility: -'.

</details>

*2 PR discussion threads considered.*

---
<details>
<summary>Completed in 1m 23s | $0.55</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 1m 23s wall · 3m 45s compute (Reviewers: 1m 01s · Synthesis: 19s) |
| Cost | $0.55 |
| Tokens | 187.5k in / 13.3k out |
| Turns | 14 |

</details>

File: tools/jtk/internal/present/comment.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/present/comment.go:181]: Intentional — the issue spec (#288) says show the restriction group/role name, not the type. Showing 'role:Administrators' adds information the spec doesn't call for.

File: tools/jtk/internal/cmd/comments/comments_test.go (hunk 2 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/comments/comments_test.go:362]: Fixed — now checks for '| - |' pattern which only matches a dash cell, not date hyphens.

File: tools/jtk/internal/cmd/comments/comments_test.go (hunk 3 above)
Comment (rianjs) [reply tools/jtk/internal/cmd/comments/comments_test.go]: Fixed — now asserts 'Visibility: -' specifically instead of bare '-'.
